src/Examine.Lucene/Indexing/FacetDateTimeType.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using Lucene.Net.Documents;
+using Lucene.Net.Facet.SortedSet;
+using Microsoft.Extensions.Logging;
+
+namespace Examine.Lucene.Indexing
+{
+    public class FacetDateTimeType : DateTimeType
```

Comment:
Wondering if it would be simpler and reduce the number of classes here to have the DateTimeType (and other underlying types that could be facetable) to have a `bool isFacetable` ctor overload. Then in the one class (i.e. DateTimeType) it checks for that value and adjusts accordingly instead of having concrete types do the faceting implementation? (hope that description makes sense :) )